int16 qr_code
---
bool ret 
